src/extensions/hooks/gp-tokens/src/actions/validate-token.ts

```diff
@@ -0,0 +1,43 @@
+import Joi, { ValidationError, type CustomHelpers } from 'joi';
+import { createError } from '@directus/errors';
+import { Token } from '../index.js';
+
+function findProtocolSymbol (str: string) {
+	// Find '://', ':/', and ':' before the first '.'
+	const firstPart = str.split('.')[0]!;
+	const regex = /(:\/\/|:\/|:)/;
+
+	const match = firstPart.match(regex);
+	return match ? match[0] : null;
+}
+
+const validateOrigin = (value: string, helpers: CustomHelpers) => {
+	const protocolSymbol = findProtocolSymbol(value);
+
+	if (protocolSymbol) {
+		value = value.replace(protocolSymbol, '://');
+	} else if (protocolSymbol === null) {
+		value = 'https://' + value;
+	}
+
+	try {
+		new URL(value);
+		return value;
+	} catch (err) {
+		return helpers.message({ custom: `Invalid URL: ${value}` });
+	}
```

Comment:
In what case would the `protocolSymbol` be just `:/` or `:` and the resulting URL end up valid?

Reply:
`new URL()` treats `:/` and `:` as valid, so I am replacing them with `://`. So `http:/jsdelivr.com` becomes `http://jsdelivr.com`. If we search only for `://` the result of `http:/jsdelivr.com` will be `https://http:/jsdelivr.com` which is still valid in the view of `new URL()`.